File: WallpaperChanger/app/src/main/java/com/saar/wallpaperchanger/Photo.java
package com.saar.wallpaperchanger;

import androidx.annotation.NonNull;

public class Photo {
    String path;
    String name;
    String artist;
    String date;

    public Photo(String path, String name, String artist) {
        this.path = path;
        this.name = name;
        this.artist = artist;

    }

    public Photo(String date, String name) {
        this.date = date;
        this.name = name;
    }

    public String getArtist() {
        return artist;
    }

    public String getPath() {
        return path;
    }

    public String getName() {
        return name;
    }

    public String getDate() {
        return date;
    }

    @NonNull
    @Override
    public String toString() {
        if (this.date != null) {
            return this.name + " - " + this.date;
        }
        return this.name + "- לא נוגן";

    }
}

File: WallpaperChanger/app/src/main/java/com/saar/wallpaperchanger/Photo.kt
package com.saar.wallpaperchanger

class Photo {
    @JvmField
    var path: String? = null
    @JvmField
    var name: String
    @JvmField
    var artist: String? = null
    var date: String? = null

    constructor(path: String?, name: String, artist: String?) {
        this.path = path
        this.name = name
        this.artist = artist
    }

    constructor(date: String?, name: String) {
        this.date = date
        this.name = name
    }

    override fun toString(): String {
        if (this.date != null) {
            return this.name + " - " + this.date
        }
        return this.name + "- לא נוגן"
    }
}
